File: API/Entities/OderAggregate/Order.cs
namespace API.Entities.OderAggregate;

    // <summary>
    /// Represents a customer order containing purchased items, payment details, and shipping information.
    /// </summary>
public class Order
{
    /// <summary>
    /// Primary key for the order entity.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// The email address of the buyer associated with this order.
    /// Used for identifying and contacting the customer.
    /// </summary>
    public required string BuyerEmail { get; set; }

    /// <summary>
    /// The shipping address where the order will be delivered.
    /// </summary>
    public required ShippingAddress ShippingAddress { get; set; }

    /// <summary>
    /// The date and time when the order was created (in UTC).
    /// Defaults to the current UTC time.
    /// </summary>
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// Collection of items included in the order.
    /// </summary>
    public List<OrderItem> OrderItems { get; set; } = [];
    
    /// <summary>
    /// The subtotal amount before delivery fees and discounts.
    /// </summary>
    public decimal Subtotal { get; set; }

     /// <summary>
    /// The fee applied for delivering the order to the specified shipping address.
    /// </summary>
    public decimal DeliveryFee { get; set; }

    /// <summary>
    /// The discount applied to the order (e.g., coupon or promo deduction).
    /// </summary>
    public decimal Discount { get; set; }

     /// <summary>
    /// The unique payment intent ID from the payment provider (e.g., Paymongo, Stripe).
    /// Used to track the payment transaction associated with this order.
    /// </summary>
    public required string PaymentIntentId { get; set; }

    /// <summary>
    /// Current status of the order (e.g., Pending, Paid, Shipped, Cancelled).
    /// </summary>
    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
    
    /// <summary>
    /// Summary of the payment details including method, reference number, or gateway metadata.
    /// </summary>
    public required PaymentSummary PaymentSummary { get; set; }


     /// <summary>
    /// Calculates the total amount due for the order after applying delivery fees and discounts.
    /// </summary>
    /// <returns>The final total amount for the order.</returns>
    public decimal GetTotal()
    {
        return Subtotal + DeliveryFee - Discount;
    }

}

File: client/src/app/models/order.ts
// Interface representing an order in the system
export interface Order {
  id: number; // Unique order ID
  buyerEmail: string; // Email of the buyer
  phone: string; // Buyer's phone number
  shippingAddress: ShippingAddress; // Shipping address details
  orderDate: string; // Date when the order was placed
  orderItems: OrderItem[]; // List of items in the order
  subtotal: number; // Total price of items before fees and discounts
  deliveryFee: number; // Delivery/shipping fee
  discount: number; // Discount applied
  total: number; // Final total amount (subtotal + deliveryFee - discount)
  orderStatus: string; // Status of the order (e.g., pending, shipped, delivered)
  paymentSummary: PaymentSummary; // Payment details for the order
}

// Interface representing a shipping address
export interface ShippingAddress {
  name: string; // Recipient's name
  line1: string; // Address line 1
  line2?: string | null; // Optional address line 2
  city: string; // City
  state: string; // State or province
  postal_code: string; // ZIP or postal code
  country: string; // Country
}

// Interface representing a single item in an order
export interface OrderItem {
  productId: number; // ID of the product
  name: string; // Product name
  pictureUrl: string; // URL of the product image
  price: number; // Price per item
  quantity: number; // Quantity ordered
}

// Interface representing payment details for an order
export interface PaymentSummary {
  last4: number | string; // Last 4 digits of the payment card
  brand: string; // Card brand (e.g., Visa, Mastercard)
  exp_month: number; // Expiration month of the card
  exp_year: number; // Expiration year of the card
}

// Interface used when creating a new order
export interface CreateOrder {
  shippingAddress: ShippingAddress; // Shipping address for the new order
  paymentSummary: PaymentSummary; // Payment details for the new order
}
